applications/minotari_node/src/stratum/tari_sha3x_stratum_handler.rs

```diff
@@ -0,0 +1,356 @@
+};
+
+#[derive(Clone)]
+pub(crate) struct TariSha3xStratumHandler<
+    TJobRepo: JobRepository + Clone + Send + Sync + 'static,
+    TBlockRepo: BlockTemplateRepository + Clone + Send + Sync + 'static,
+> {
+    block_template_repository: TBlockRepo,
+    job_repository: TJobRepo,
+    submit_job_queue_sender: SubmitJobQueueSender,
+}
+
+impl<
+        TJobRepo: JobRepository + Clone + Send + Sync + 'static,
+        TBlockRepo: BlockTemplateRepository + Clone + Send + Sync + 'static,
+    > TariSha3xStratumHandler<TJobRepo, TBlockRepo>
+{
+    pub(crate) fn new(
+        block_template_repository: TBlockRepo,
+        job_repository: TJobRepo,
+        submit_job_queue_sender: SubmitJobQueueSender,
+    ) -> Self {
+        Self {
+            block_template_repository,
+            job_repository,
+            submit_job_queue_sender,
+        }
+    }
+}
+
+#[async_trait]
+impl<
+        TJobRepo: JobRepository + Clone + Send + Sync + 'static,
+        TBlockRepo: BlockTemplateRepository + Clone + Send + Sync + 'static,
+    > StratumJobHandler for TariSha3xStratumHandler<TJobRepo, TBlockRepo>
+{
+    async fn login(
+        &self,
+        _id: String,
+        _login: String,
+        address: String,
+        algo: &[String],
+    ) -> anyhow::Result<LoginResponse> {
+        // Handle login request
+        let address = TariAddress::from_str(&address)?;
+
+        let main_algo;
+        if algo.is_empty() {
+            main_algo = "cuckaroo".to_string();
+        } else if algo.len() == 1 {
+            main_algo = algo[0].clone();
+        } else if algo.iter().any(|a| a.as_str() == "rx/0") {
+            main_algo = "RandomXT".to_string();
+        } else {
+            return Err(anyhow::anyhow!("Unsupported algorithm: {}", algo.join(", ")));
+        }
+
+        let algo = main_algo.parse()?;
+        let BlockTemplate {
+            blob,
+            height,
+            target,
+            seed_hash,
+            prev_block_hash,
+            ..
+        } = self
+            .block_template_repository
+            .get_block_template(algo, Some(address.clone()))
+            .await?;
+        let mut r = OsRng;
+        let job_id = hex::encode(r.next_u64().to_le_bytes());
+
+        let id = hex::encode(r.next_u64().to_le_bytes());
+        let random_bytes = rand::random::<u16>();
+        let job_target = hex::encode((u64::MAX / target).to_le_bytes());
+        let chain_target = hex::encode(target.to_le_bytes());
+
+        if blob.is_empty() {
+            return Err(anyhow::anyhow!("No blob available for the latest block"));
+        }
+        let original_mining_hash = blob.clone();
+        let blob = if main_algo == "RandomXT" {
+            // The format is:
+            // | 1 byte | 1 byte | 1 bytes | 32 bytes | 8 bytes | 1 byte | 32 bytes|
+            // | major version | minor version | timestamp | mining_hash | nonce (big endian) | pow_algo | pow_data, excluding algo, padded to 32 bytes |
```

Comment:
Easier to read if we align them
```suggestion
        // The format is:
        // | 1 byte        | 1 byte        | 1 bytes   | 32 bytes    | 8 bytes            | 1 byte   | 32 bytes                                     |
        // | major version | minor version | timestamp | mining_hash | nonce (big endian) | pow_algo | pow_data, excluding algo, padded to 32 bytes |
```